src/dynimage.rs

```diff
@@ -194,110 +194,306 @@ impl DynamicImage {
     ///
     /// If the image was already the correct format, it is returned as is.
     /// Otherwise, a copy is created.
+    #[deprecated = "replaced by `into_rgb8`"]
     pub fn into_rgb(self) -> RgbImage {
         match self {
             DynamicImage::ImageRgb8(x) => x,
-            x => x.to_rgb(),
+            x => x.to_rgb8(),
+        }
+    }
+
+    /// Consume the image and returns a RGB image.
+    ///
+    /// If the image was already the correct format, it is returned as is.
+    /// Otherwise, a copy is created.
+    pub fn into_rgb8(self) -> RgbImage {
+        match self {
+            DynamicImage::ImageRgb8(x) => x,
+            x => x.to_rgb8(),
+        }
+    }
+
+    /// Consume the image and returns a RGB image.
+    ///
+    /// If the image was already the correct format, it is returned as is.
+    /// Otherwise, a copy is created.
+    pub fn into_rgb16(self) -> Rgb16Image {
+        match self {
+            DynamicImage::ImageRgb16(x) => x,
+            x => x.to_rgb16(),
         }
     }
 
     /// Consume the image and returns a RGBA image.
     ///
     /// If the image was already the correct format, it is returned as is.
     /// Otherwise, a copy is created.
+    #[deprecated = "replaced by `into_rgba8`"]
     pub fn into_rgba(self) -> RgbaImage {
         match self {
             DynamicImage::ImageRgba8(x) => x,
-            x => x.to_rgba(),
+            x => x.to_rgba8(),
+        }
+    }
+
+    /// Consume the image and returns a RGBA image.
+    ///
+    /// If the image was already the correct format, it is returned as is.
+    /// Otherwise, a copy is created.
+    pub fn into_rgba8(self) -> RgbaImage {
+        match self {
+            DynamicImage::ImageRgba8(x) => x,
+            x => x.to_rgba8(),
+        }
+    }
+
+    /// Consume the image and returns a RGBA image.
+    ///
+    /// If the image was already the correct format, it is returned as is.
+    /// Otherwise, a copy is created.
+    pub fn into_rgba16(self) -> Rgba16Image {
+        match self {
+            DynamicImage::ImageRgba16(x) => x,
+            x => x.to_rgba16(),
         }
     }
 
     /// Consume the image and returns a BGR image.
     ///
     /// If the image was already the correct format, it is returned as is.
     /// Otherwise, a copy is created.
+    #[deprecated = "replaced by `into_bgra8`"]
     pub fn into_bgr(self) -> BgrImage {
         match self {
             DynamicImage::ImageBgr8(x) => x,
-            x => x.to_bgr(),
+            x => x.to_bgr8(),
+        }
+    }
+
+    /// Consume the image and returns a BGR image.
+    ///
+    /// If the image was already the correct format, it is returned as is.
+    /// Otherwise, a copy is created.
+    pub fn into_bgr8(self) -> BgrImage {
+        match self {
+            DynamicImage::ImageBgr8(x) => x,
+            x => x.to_bgr8(),
         }
     }
 
+    // NOTE: DynamicImage::ImageBgr16 variant does not currently exist, so there is no `into_bgr16`
```

Comment:
This variant will probably never exist. It is kind of unfortunate that we have BGR at all, but adding it in more places would probably require a strong reason beyond just being consistent.

Reply:
At least it serves as a reminder for any plan of a replacement that this needs to be considered early. I wouldn't change the comment actually.